### ISSUE TITLE
x/interchainstaking/keeper: needless scaling and truncating of weights in AggregateIntents

### ISSUE THREAD

**Issue by elias-orijtech** [ingenuity-build/quicksilver#200]
`keeper.AggregateIntents` scales all intent weights by the zone redemption rate:

https://github.com/ingenuity-build/quicksilver/blob/8c558f01b2b2528bd36c484da2c0b5d64e2b110b/x/interchainstaking/keeper/intent.go#L114

However, because weights end up a fraction of the total weight, the scaling has no effect.

The subsequent truncation (`TruncateInt`) also seems needlessly confusing. Why not keep weights in `sdk.Dec` and add a `DelegatorIntent.Ordinalize` variant that takes a `sdk.Dec`?

In other words, removing scale and truncation makes the intent distribution correctness easier to verify.

CC @odeke-em

**Comment by elias-orijtech** [comment]
The same comment applies to `keeper.UpdateIntent`:

https://github.com/ingenuity-build/quicksilver/blob/8c558f01b2b2528bd36c484da2c0b5d64e2b110b/x/interchainstaking/keeper/intent.go#L153

and increases the motivation for simplifying the intent weights: it's not trivial to check whether the redemtion rate stays constant between calls to `AggregateIntents` and `UpdateIntent`.

**Comment by zanicar** [comment]
This was addressed in #235?

**Comment by elias-orijtech** [comment]
I still see the issue in UpdateIntent

https://github.com/ingenuity-build/quicksilver/blob/20dc658480b1af1cb323b4ab4a8e5925ee79a0ed/x/interchainstaking/keeper/intent.go#L168

?

**Comment by joe-bowman** [comment]
No it wasn't addressed.

Looking at this again @elias-orijtech - I'm not sure I agree with the problem here?

We raise each user by their balance - the balance of each account is different. The different weighting matter:
```
user A gives validator 0.5 and has balance of 10
user B gives validator 0.25 and has balance of 5

current calc:  ((0.5 * 10) + (0.25 * 5)) / 15  == 0.41666
without scaling: == (0.5 + 0.25) / 2 == 0.3333
```

**Comment by elias-orijtech** [comment]
Perhaps we're talking past each other. I don't think I'm objecting to weighing as you describe it, but to the constant scaling (and truncation) of weights. So this issue is about replacing

```
		baseBalance := zone.RedemptionRate.Mul(sdk.NewDecFromInt(balance.Balance.Amount)).TruncateInt()
```

with

```
		baseBalance := sdk.NewDecFromInt(balance.Balance.Amount)
```

because `zone.RedemptionRate` is constant across all weights, and TruncateInt complicates the calculation.

Note that I'm not saying the code is wrong, merely that it can be simplified to be easier to review.

**Comment by joe-bowman** [comment]
Thank you for the update. I blame the last post on jetlag :)

So looking at the individual functions, the redemption rate _does_ matter in the terms of `UpdateIntent` and does not in `AggregateIntents`.

With `AggregateIntents` we are scaling up and down in the same method; so the scaling was removed in #235.

However, for `UpdateIntent` we are adding values that are already ordinal with respect to the redemption rate (`inAmount`). 

I am adding some additional docs and tests to clarify this.
